src/components/clusters/newClusterModal.tsx

```diff
@@ -47,9 +47,10 @@ export const NewClusterModalButton: React.FC<NewClusterModalButtonProps> = ({
 
 type NewClusterModalProps = {
   closeModal: () => void;
+  history: History;
 };
 
-export const NewClusterModal: React.FC<NewClusterModalProps> = ({ closeModal }) => {
+export const NewClusterModal: React.FC<NewClusterModalProps> = ({ closeModal, history }) => {
```

Comment:
why dont we use withHistory HOC ?
Please also note that UHC uses older router (v4) which does not have hooks so we should avoid using them. Maybe we should downgrade the router version to make sure we wont use any new features of v5. v5 and v4 is backward compatibile so we shouldnt have any problems doing so.

Reply:
I didn't see it necessary to use the HOC when the props passing is just one level.

Reply:
We can downgrade the router in a separate PR